Issue 92

yegawa opened the pull request.

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

とりあえず淡白にカレンダーに時間、人の名前、実験名を表示するようにした。
TODO
・現状ボタンである必要はないのでテキストに変更
・submitボタンも不要なので変更
・どっかにcalendar_init.js的なのを作ってapplicationsと統一する

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

・ボタンをテキストに変更
・submitボタンを削除
TODO
・カレンダー表示部分をcalendar_initにまとめる。

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

calendar_init.js作成。
引き続きテスト書きます。

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

applications_controller.rbの不要行削除

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

js側の細かいテストはできていませんが、せめてインスタンス変数のチェックをと思ってそれだけやりました。

okabi commented on app/controllers/members_controller.rb:
```diff
@@ -23,6 +23,22 @@ def create
+    @schedules.each do |s|
+      participant = ''
+      if s.participant
+        participant = s.participant.name
```
#88 がマージされると `Schedule` から `participant` 消えるのでこれだと動かないんだけど、
どっちが合わせるのが良いんかな？

okabi commented on app/controllers/members_controller.rb:
```diff
@@ -23,6 +23,22 @@ def create
+      @times.push({
+        start: s.datetime,
+        end: s.datetime + s.experiment.duration * 60,
+        experiment: @experiments.find_by_id(s.experiment_id).name,
```
`experiment: s.experiment.name` の方が見た目分かりやすいかな？
たぶん動作は同じだけど。

okabi reviewed (commented):
`ids` とか差集合演算とか参考になりました 😄

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

とりあえずrebase前にできてたスケジュールの表示のところまで。
引き続きstatus: 0に対する処理（単純に一人のとき、バッティングのときに云々）を書く

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

1つのscheduleに複数人いた場合の処理書いた。
癖でjsだよりなコードになってしまった。

okabi commented on app/controllers/members_controller.rb:
```diff
@@ -23,6 +23,29 @@ def create
+    @schedules.each do |s|
+      p_infos = []
+      my_apps = Application.where(schedule_id: s.id).where(status: [0, 1])
+      if !my_apps.empty?
```
この `empty?` って要るのかな？
空配列でも `each`  で処理できる気がする

okabi reviewed (approved):
ざっと見た感じ大丈夫だと思います！

yegawa committed:
[ref #92] [member] マイページに実験予定を表示する

不要なif文の削除。
